unit-tests/post-processing/test-post-processing-filters.py

```diff
@@ -0,0 +1,469 @@
+    # Downsample scales 2 and 3 are tested only. scales 4-7 are differ in impementation from the reference code:
+    # In Librealsense for all scales [2-8] the filter is the mean of depth.
+    # I the reference code for [2-3] the filter uses mean of depth, and for 4-7 is switches to man of disparities doe to implementation constrains
+    ("1551257764229", "D435_DS(2)"),
+    ("1551257812956", "D435_DS(3)"),
+    # Downsample + Hole-Filling modes 0/1/2
+    ( "1551257880762","D435_DS(2)_HoleFill(0)" ),
+    ( "1551257882796","D435_DS(2)_HoleFill(1)" ),
+    ( "1551257884097","D435_DS(2)_HoleFill(2)" ),
+    # Downsample + Spatial Filter parameters
+    ( "1551257987255",  "D435_DS(2)+Spat(A:0.85/D:32/I:3)" ),
+    ( "1551259481873",  "D435_DS(2)+Spat(A:0.5/D:15/I:2)" ),
+    #Downsample + Temporal Filter
+    ( "1551261946511",  "D435_DS(2)+Temp(A:0.25/D:15/P:0)" ),
+    ( "1551262153516",  "D435_DS(2)+Temp(A:0.45/D:25/P:1)" ),
+    ( "1551262256875",  "D435_DS(2)+Temp(A:0.5/D:30/P:4)" ),
+    ( "1551262841203",  "D435_DS(2)+Temp(A:0.5/D:30/P:6)" ),
+    ( "1551262772964",  "D435_DS(2)+Temp(A:0.5/D:30/P:8)" ),
+    # Downsample + Spatial + Temporal
+    ( "1551262971309",  "D435_DS(2)_Spat(A:0.7/D:25/I:2)_Temp(A:0.6/D:15/P:6)" ),
+    # Downsample + Spatial + Temporal (+ Hole-Filling)
+    ( "1551263177558",  "D435_DS(2)_Spat(A:0.7/D:25/I:2)_Temp(A:0.6/D:15/P:6))_HoleFill(1)" ),
+]
+
+metadata_attributes = {
+    'res_x':        "Resolution_x",
+    'res_y':        "Resolution_y",
+    'focal_length':  "Focal Length",
+    'depth_units':   "Depth Units",
+    'stereo_baseline':"Stereo Baseline",
+    'downscale':    "Scale",
+    'spat_filter':   "Spatial Filter Params:",
+    'spat_alpha':   "SpatialAlpha",
+    'spat_delta':   "SpatialDelta",
+    'spat_iter':    "SpatialIterations",
+    'temp_filter':  "Temporal Filter Params:",
+    'temp_alpha':   "TemporalAlpha",
+    'temp_delta':   "TemporalDelta",
+    'temp_persist': "TemporalPersistency",
+    'holes_filter': "Holes Filling Mode:",
+    'holes_fill':   "HolesFilling",
+    'frames_sequence_size': "Frames sequence length"
+}
+class post_proccesing_filters:
+    def __init__(self):
+        self.depth_to_disparity = rs.disparity_transform(True)
+        self.disparity_to_depth = rs.disparity_transform(False)
+        self.dec_pb = False
+        self.spat_pb = False
+        self.temp_pb = False
+        self.holes_pb = False
+        self.dec_filter = rs.decimation_filter()
+        self.spat_filter = rs.spatial_filter()
+        self.temp_filter = rs.temporal_filter()
+        self.hole_filling_filter = rs.hole_filling_filter()
+
+    def configure(self,filters_cfg):
+        #Reconfigure the post-processing according to the test spec
+        self.dec_pb = filters_cfg.downsample_scale != 1
+        self.dec_filter.set_option(rs.option.filter_magnitude, filters_cfg.downsample_scale)
+
+        if filters_cfg.spatial_filter:
+            self.spat_pb = filters_cfg.spatial_filter
```

Comment:
If `configured` was called with `filters_cfg.spatial_filter == True` and then a second time with `filters_cfg.spatial_filter == False` shouldn't we set this? Same for `temp_pb` and `holes_pb`

Reply:
Fixed